lib/newrelic/elasticsearch.rb

```diff
@@ -19,22 +19,28 @@
     NewRelic::Agent::MethodTracer.extend(NewRelic::Agent::MethodTracer)
 
     ::Elasticsearch::Transport::Client.class_eval do
-      def perform_request_with_new_relic(method, path, params={}, body=nil)
+      @@old_version = method(:perform_request).parameters.length == 4
+      def perform_request_with_new_relic(method, path, params={}, body=nil, headers=nil)
         resolver = NewRelic::ElasticsearchOperationResolver.new(method, path)
 
         callback = proc do |result, metric, elapsed|
-          statement = { body: body, params: params }
+          statement = { body: body, params: params , headers: headers}
           statement[:scope] = resolver.scope
           statement[:additional_parameters] = resolver.operands
 
           NewRelic::Agent::Datastores.notice_statement(statement.inspect, elapsed) if statement
         end
 
         NewRelic::Agent::Datastores.wrap('Elasticsearch', resolver.operation_name, resolver.index, callback) do
-          perform_request_without_new_relic(method, path, params, body)
+          if @@old_version
```

Comment:
This is probably not the right way to do this.   Pretty sure the Elasticsearch gem we're monkey patching has a VERSION constant.   Interrogate that constant with `Gem::Version` from the Stdlib and I think we can merge this.

Reply:
Yeah I think in this case we could reference `::Elasticsearch::Transport::VERSION`. This would make it clearer what "old version" actually means.

Could also consider moving the conditional into the metaprogramming level to avoid having to check the boolean on every single elasticsearch query.

Reply:
With regards to thread safety you can't have more than one gem version installed at a time so the value of `@@old_version` will never mutate (ie. it's thread safe).